e2e/testcafe-devextreme/tests/dataGrid/keyboardNavigation/editOnKeyPress.screenshots.ts

```diff
@@ -62,3 +65,55 @@ const DATA_GRID_SELECTOR = '#container';
     await makeRowsViewTemplatesAsync(DATA_GRID_SELECTOR);
   });
 });
+
+test('Focused cell should not flick', async (t) => {
+  const dataGrid = new DataGrid(DATA_GRID_SELECTOR);
+  const firstCell = dataGrid.getDataCell(0, 0).element;
+  const secondCell = dataGrid.getDataCell(1, 0).element;
+  await t.click(firstCell);
+  await t.click(secondCell);
+
+  await ClientFunction(() => {
+    // @ts-expect-error
+    window.counter = 0;
+    (secondCell() as any as HTMLElement).addEventListener('focusin', () => {
+      // @ts-expect-error
+      window.counter += 1;
+    });
+  }, {
+    dependencies: { secondCell },
+  })();
+
+  await t.pressKey('M');
+  await t.pressKey('enter');
+
+  const focusEventCount = await ClientFunction(
+    // @ts-expect-error
+    () => window.counter,
+  )();
+
+  await t.expect(focusEventCount).eql(1);
+}).before(async () => {
+  await createWidget('dxDataGrid', {
+    dataSource: [
+      {
+        value: 'data',
+      },
+      {
+        value: 'data',
+      },
+    ],
```

Comment:
This issue isn't reproduces with array local dataSource :(

So, we should implement a custom store here I guess.
Something like:

```javascript
const data = [
                {
                    Id: 0,
                    A: 'A_0',
                    B: 'B_0',
                    C: 'C_0',
                },
                {
                    Id: 1,
                    A: 'A_1',
                    B: 'B_1',
                    C: 'C_1',
                },
                {
                    Id: 2,
                    A: 'A_2',
                    B: 'B_2',
                    C: 'C_2',
                }
            ];
            const update = {
                Id: 100,
                A: 'A_UPDATE',
                B: 'B_UPDATE',
                C: 'C_UPDATE',
            }

            const ordersStore = new DevExpress.data.CustomStore({
                key: 'Id',
                load() {
                    return Promise.resolve(data)
                },
                update() {
                    let resolve = null;
                    const promise = new Promise((r) => { resolve = r });

                    setTimeout(() => {
                        resolve(update);
                    }, 100);

                    return promise;
                }
            });
```